File: app/src/main/java/com/stathis/elmepaunivapp/ui/research/ResearchActivity.java
package com.stathis.elmepaunivapp.ui.research;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.RecyclerView;
import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.stathis.elmepaunivapp.R;
import com.stathis.elmepaunivapp.abstraction.AbstractActivity;
import com.stathis.elmepaunivapp.listeners.activity_listeners.ResearchActivityClickListener;
import com.stathis.elmepaunivapp.ui.students.model.UsefulLinks;
import com.stathis.elmepaunivapp.ui.dashboard.DashboardActivity;
import com.stathis.elmepaunivapp.ui.professors.ProfessorsActivity;
import com.stathis.elmepaunivapp.ui.students.StudentsActivity;
import com.stathis.elmepaunivapp.ui.webview.WebviewActivity;

public class ResearchActivity extends AbstractActivity implements ResearchActivityClickListener {

    private RecyclerView researchRecycler;
    private BottomNavigationView bottomNavigationView;
    private ResearchViewModel viewModel;

    public ResearchActivity() {
        super(R.layout.activity_research_in_dept);
    }

    @Override
    public void initial() {
        viewModel = new ViewModelProvider(this).get(ResearchViewModel.class);

        researchRecycler = findViewById(R.id.research_activity_recycler);
        researchRecycler.setAdapter(viewModel.researchFinalAdapter);
        bottomNavigationView= findViewById(R.id.bottom_nav);
    }

    @Override
    public void running() {
        viewModel.initListener(this);
        viewModel.createLists();
        viewModel.displayData();

        bottomNavigationView.setSelectedItemId(R.id.nav_uni);
        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                switch (item.getItemId()) {
                    case R.id.nav_home:
                        startActivity(new Intent(ResearchActivity.this, DashboardActivity.class));
                        break;
                    case R.id.nav_students:
                        startActivity(new Intent(ResearchActivity.this, StudentsActivity.class));
                        break;
                    case R.id.nav_uni:
                        return true;
                    case R.id.nav_search:
                        startActivity(new Intent(ResearchActivity.this, ProfessorsActivity.class));
                        break;
                }
                return false;
            }
        });
    }

    @Override
    public void stopped() {

    }

    @Override
    public void itemClicked(UsefulLinks usefulLinks) {
        startActivity(new Intent(ResearchActivity.this, WebviewActivity.class).putExtra("URL", usefulLinks.getUrl()));
    }
}

File: app/src/main/java/com/stathis/elmepaunivapp/ui/research/ResearchActivity.kt
package com.stathis.elmepaunivapp.ui.research

import android.content.Intent
import androidx.activity.viewModels
import com.stathis.elmepaunivapp.R
import com.stathis.elmepaunivapp.abstraction.ElmepaActivity
import com.stathis.elmepaunivapp.callbacks.ResearchClickListener
import com.stathis.elmepaunivapp.databinding.ActivityResearchInDeptBinding
import com.stathis.elmepaunivapp.ui.research.model.ResearchItem
import com.stathis.elmepaunivapp.ui.webview.WebviewActivity
import com.stathis.elmepaunivapp.util.TITLE
import com.stathis.elmepaunivapp.util.URL
import com.stathis.elmepaunivapp.util.setupBar
import kotlinx.android.synthetic.main.activity_research_in_dept.*

class ResearchActivity : ElmepaActivity<ActivityResearchInDeptBinding>(R.layout.activity_research_in_dept) {

    private val viewModel: ResearchViewModel by viewModels()

    override fun init() {
        supportActionBar?.setupBar(getString(R.string.research_title))
        binding.viewModel = viewModel
    }

    override fun startOps() {
        viewModel.observe(this, object : ResearchClickListener {
            override fun onItemTap(item: ResearchItem) {
                startActivity(Intent(this@ResearchActivity, WebviewActivity::class.java).apply {
                    putExtra(URL, item.url)
                    putExtra(TITLE, item.name)
                })
            }
        })
    }

    override fun stopOps() = viewModel.release(this)

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }
}